src/main/java/com/extendedclip/deluxemenus/persistentmeta/PersistentMetaHandler.java

```diff
@@ -262,7 +262,18 @@ public <T> Map<String, T> getMetaValues(
             return OperationResult.SUCCESS;
         }
 
-        final long newValue = (long) (currentValue == null ? 0 : currentValue) + value.longValue();
+        long currentValueLong = 0;
+        if (currentValue != null) {
+            if (currentValue instanceof Long) {
+                currentValueLong = (Long) currentValue;
+            } else if (currentValue instanceof Integer) {
+                currentValueLong = (Integer) currentValue;
+            } else {
+                return OperationResult.EXISTENT_VALUE_IS_DIFFERENT_TYPE;
+            }
+        }
+
+        final long newValue = currentValueLong + (type == DataType.INTEGER ? value.intValue() : value.longValue());

```

Comment:
I was unaware that a cast to `long` from `Integer` and `Double` would fail.

But since we know these values can only be Integer, Long and Double, maybe this code can be simplified a bit more:

```java
final long newValue = ((Number) (currentValue == null ? 0 : currentValue)).longValue() + value.longValue();
```

Integer, Long and Double, all extend Number. And number has this nice longValue() method we can use.